google/cloud/bigquery_v2/services/centralized_service/client.py

```diff
@@ -43,139 +45,198 @@
 from google.api_core import retry as retries
 from google.auth import credentials as auth_credentials
 
-# Create a type alias
+# Create type aliases
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
 
-# TODO: This line is here to simplify prototyping, etc.
-PROJECT_ID = os.environ.get("GOOGLE_CLOUD_PROJECT")
+DatasetIdentifier = Union[str, dataset_reference.DatasetReference]
 
+# TODO: This variable is here to simplify prototyping, etc.
+PROJECT_ID = os.environ.get("GOOGLE_CLOUD_PROJECT")
 DEFAULT_RETRY: OptionalRetry = gapic_v1.method.DEFAULT
 DEFAULT_TIMEOUT: Union[float, object] = gapic_v1.method.DEFAULT
 DEFAULT_METADATA: Sequence[Tuple[str, Union[str, bytes]]] = ()
 
-
 # Create Centralized Client
 class BigQueryClient:
+    """A centralized client for BigQuery API."""
+
     def __init__(
         self,
         *,
         credentials: Optional[auth_credentials.Credentials] = None,
         client_options: Optional[Union[client_options_lib.ClientOptions, dict]] = None,
     ):
-        self._clients = {}
+        """
+        Initializes the BigQueryClient.
+
+        Args:
+            credentials:
+                The credentials to use for authentication. If not provided, the
+                client will attempt to use the default credentials.
+            client_options:
+                A dictionary of client options to pass to the underlying
+                service clients.
+        """
+
+        self._clients: Dict[str, object] = {}
         self._credentials = credentials
         self._client_options = client_options
+        self.project = PROJECT_ID
+
+    # --- HELPER METHODS ---
+    def _parse_dataset_path(self, dataset_path: str) -> Tuple[Optional[str], str]:
+        """
+        Helper to parse project_id and/or dataset_id from a string identifier.
+
+        Args:
+            dataset_path: A string in the format 'project_id.dataset_id' or
+                'dataset_id'.
+
+        Returns:
+            A tuple of (project_id, dataset_id).
+        """
+        if "." in dataset_path:
+            project_id, dataset_id = dataset_path.split(".", 1)
+            return project_id, dataset_id
+        return self.project, dataset_path
+
+    def _parse_dataset_id_to_dict(self, dataset_id: DatasetIdentifier) -> dict:
+        """
+        Helper to create a request dictionary from a project_id and dataset_id.
 
+        Args:
+            dataset_id: A string or DatasetReference.
+
+        Returns:
+            A dict of {"project_id": project_id, "dataset_id": dataset_id_str }.
+        """
+        if isinstance(dataset_id, str):
+            project_id, dataset_id_str = self._parse_dataset_path(dataset_id)
+            return {"project_id": project_id, "dataset_id": dataset_id_str}
+        elif isinstance(dataset_id, dataset_reference.DatasetReference):
```

Comment:
In this case, can't we use the https://googleapis.dev/python/protobuf/latest/google/protobuf/json_format.html#google.protobuf.json_format.MessageToDict or the proto-plus equivalent?